osu.Game/Extensions/ModelExtensions.cs

```diff
@@ -137,20 +137,14 @@ private static bool matchesOnlineID(this IHasOnlineID<int>? instance, IHasOnline
             return instance.OnlineID.Equals(other.OnlineID);
         }
 
-        private static readonly char[] invalid_filename_characters = Path.GetInvalidFileNameChars()
-                                                                         // Backslash is added to avoid issues when exporting to zip.
-                                                                         // See SharpCompress filename normalisation https://github.com/adamhathcock/sharpcompress/blob/a1e7c0068db814c9aa78d86a94ccd1c761af74bd/src/SharpCompress/Writers/Zip/ZipWriter.cs#L143.
-                                                                         .Append('\\')
-                                                                         .ToArray();
+        private static Regex invalid_filename_chars = new(@"(?!$)[^A-Za-z0-9_()[\] -]", RegexOptions.Compiled);
```

Comment:
I believe that `(?!$)[^A-Za-z0-9_()[\] -\\.]` includes the . character as well.

Reply:
Yeah, changed it. Should be careful to not make it a range ("` -.`" would match any character between space and `.`).